### ISSUE TITLE
support converting wx.Bitmap, wx.Icon and wx.Image to wx.BitmapBundle automatically

### ISSUE THREAD

**Issue by jmoraleda** [wxWidgets/Phoenix#2153]
It is my understanding that pure C++ applications use the conversion constructors of `wx.BitmapBundle` (https://docs.wxwidgets.org/latest/classwx_bitmap_bundle.html#a922b10aa1d1127d38a169fc2281a2e03) to automatically create single bitmat bundles when a `wx.Bitmap`, `wx.Icon` or `wx.Image` is passed to a method or constructor that expects a `wx.BitmapBundle`.

This feature means that older C++ applications work seamlessly after the upgrade to `3.1.6`. Unfortunately this is not true for wxPython applications where the conversion does not currently happen automatically. This means that an update of wxPython breaks older applications.

I am wondering what the thoughts are on adding this feature to wxPython.

I remark that updating older applicatons using `BitmapBundle.FromBitmap` where required is a straightforward, but things get more interesting when using autogenerated code from, for example, wxFormBuilder. I recently encountered this issue and have been having a discussion there that is relevant here: https://github.com/wxFormBuilder/wxFormBuilder/discussions/739

So in summary. What are the thoughts? Should wxPython support automatic conversion or should code generating applications migrate to the new API?

**Comment by efahl** [comment]
It would be nice to get a user-transparent fix for this, as it appears to affect a pretty wide collection of code.  After playing whack-a-mole on my own code (about 10 'SetBitmapX' and 'AddTool' calls), I ran into one in matplotlib (v 3.5.2).  I can only assume there are more...

```
  File "Python310/lib/site-packages/matplotlib/backends/backend_wx.py", line 965, in _get_toolbar
    toolbar = NavigationToolbar2Wx(self.canvas)
  File "Python310/lib/site-packages/matplotlib/backends/backend_wx.py", line 1109, in __init__
    self.AddTool(
TypeError: ToolBar.AddTool(): arguments did not match any overloaded call:
  overload 1: argument 1 has unexpected type 'int'
  overload 2: argument 'bitmap' has unexpected type 'Bitmap'
  overload 3: argument 'bitmap' has unexpected type 'Bitmap'
```

**Comment by FreeHarry** [comment]
According wxpython documentation converting should be done automatically ...

> Also note that the existing code using [wx.Bitmap](https://wxpython.org/Phoenix/docs/html/wx.Bitmap.html#wx-bitmap) is compatible with the functions taking wx.BitmapBundle in wxWidgets 3.1.6 and later because bitmaps are implicitly convertible to the objects of this class, so just passing wx.Bitmap to the functions taking [wx.BitmapBundle](https://wxpython.org/Phoenix/docs/html/wx.BitmapBundle.html#wx-bitmapbundle) continues to work and if high resolution versions of bitmap are not (yet) available for the other toolbar tools, single bitmaps can continue to be used instead.

See within the description of [wx.BitmapBundle](https://wxpython.org/Phoenix/docs/html/wx.BitmapBundle.html#phoenix-title-wx-bitmapbundle).

**Comment by swt2c** [comment]
> According wxpython documentation converting should be done automatically ...
> 
> > Also note that the existing code using [wx.Bitmap](https://wxpython.org/Phoenix/docs/html/wx.Bitmap.html#wx-bitmap) is compatible with the functions taking wx.BitmapBundle in wxWidgets 3.1.6 and later because bitmaps are implicitly convertible to the objects of this class, so just passing wx.Bitmap to the functions taking [wx.BitmapBundle](https://wxpython.org/Phoenix/docs/html/wx.BitmapBundle.html#wx-bitmapbundle) continues to work and if high resolution versions of bitmap are not (yet) available for the other toolbar tools, single bitmaps can continue to be used instead.
> 
> See within the description of [wx.BitmapBundle](https://wxpython.org/Phoenix/docs/html/wx.BitmapBundle.html#phoenix-title-wx-bitmapbundle).

Yeah...but that's just autogenerated coming from the wxWidgets interface files.  Doesn't mean that it is automatically true for wxPython.  :-)

**Comment by jmoraleda** [comment]
I am closing the issue because I believe Robin's commit https://github.com/wxWidgets/Phoenix/commit/c054d18112c5fc7e5bee91a76a50fc1172768f9b resolves it: Per that commit wxPython now also automatically converts to `Bitmap`, `Icon` and `Image` to `BitmapBundle`
